tests/analyses/decompiler/test_decompiler.py

```diff
@@ -2898,6 +2898,131 @@ def test_multistatementexpression_od_read_char(self, decompiler_options=None):
         self._print_decompilation_result(dec)
         assert dec.codegen.text == saved
 
+    @structuring_algo("phoenix")
+    def test_ifelseflatten_iplink_bridge(self, decompiler_options=None):
+        bin_path = os.path.join(test_location, "x86_64", "decompiler", "iplink_bridge.o")
+        proj = angr.Project(bin_path, auto_load_libs=False)
+        cfg = proj.analyses.CFGFast(normalize=True, data_references=True)
+
+        f = proj.kb.functions["bridge_print_opt"]
+        proj.analyses.CompleteCallingConventions(cfg=cfg, recover_variables=True, analyze_callsites=True)
+        d = proj.analyses[Decompiler](f, cfg=cfg.model, options=decompiler_options)
+
+        self._print_decompilation_result(d)
+        text = d.codegen.text
+        good_if_return = "if (!a2)\n        return 0;\n"
+        first_if_location = text.find("if")
+
+        # TODO: this is broken right now on the 1 goto for a bad else. It may not be relevant for this testcase.
```

Comment:
If you think this is out of scope, we can just remove it, since it may not have anything to do with if-flipping